articles/css-grid/index.md

```diff
@@ -0,0 +1,379 @@
+  <div class="grid-child-4 grid-child">Grid Child 4</div>
+</div>
+```
+
+The above code creates four columns of one fraction space each. In this case, they take up 25% space each.
+
+### grid-template-areas
+
+This property defines the grid template. It defines which page elements are placed in which grid in the layout.
+
+```css
+.child-1 {
+  grid-area: area-1;
+}
+
+.child-2 {
+  grid-area: area-2;
+}
+
+.child-3 {
+  grid-area: area-3;
+}
+
+.grid-parent-container {
+  display: grid;
+  grid-template-columns: 1fr 1fr 1fr;
+  grid-template-rows: 1fr 1fr;
+  grid-template-areas:
+    "area-1 area-2 area-2"
+    "area-1 area-3 area-3";
+}
+```
+
+The above snippet will create a grid that is three columns wide and two rows tall. The top row will be comprised of area-1 and two cells of area-2. The bottom row will be composed of a cell of area-1 and two cells of area-3.
+
+The syntax of `grid-template-areas` provides a visualization of the defined grid structure.
+
+_Note: There is more terminology related to CSS Grid. We have covered only the important ones so that we get a basic understanding of the grid features. For a more detailed overview of all grid features, visit this [MDN link](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout)_
+
+## Let's Code
+
+Our goal is to generate a simple image grid as shown.
+
+![Image Grid](/engineering-education/css-grid/image-grid.png)
+
+Let us start by defining our page's HTML.
+
+```html
+<div class="grid-parent-container">
+  <div class="grid-child small-img1">
+    <img src="https://source.unsplash.com/random/200x100" />
+  </div>
+  <div class="grid-child medium-img1">
+    <img src="https://source.unsplash.com/random/200x400" />
+  </div>
+  <div class="grid-child large-img1">
+    <img src="https://source.unsplash.com/random/300x600" />
+  </div>
+  <div class="grid-child medium-img2">
+    <img src="https://source.unsplash.com/random/210x400" />
+  </div>
+  <div class="grid-child small-img2">
+    <img src="https://source.unsplash.com/random/210x100" />
+  </div>
+  <div class="grid-child large-img2">
+    <img src="https://source.unsplash.com/random/400x600" />
+  </div>
+</div>
+```
+
+Here, we have the `grid-parent-container`. This is the wrapper for the grid layout we made. The images are fetched using a random image generator, [Unsplash](https://unsplash.com/).
+
+Let us write the CSS and define our grid layout.
+
+```css
+/* The grid-area defines the names of the grid areas that would be speicified 
+in the grid-template-areas in the parent container */
+.small-img1 {
+  grid-area: small-img1;
+}
+.medium-img1 {
+  grid-area: medium-img1;
+}
+.large-img1 {
+  grid-area: large-img1;
+}
+.medium-img2 {
+  grid-area: medium-img2;
+}
+.small-img2 {
+  grid-area: small-img2;
+}
+.large-img2 {
+  grid-area: large-img2;
+}
+
+/* Parent Grid Container */
+.grid-parent-container {
+  display: grid;
+  justify-items: center; /* Centers the grid items on the row axis */
```

Comment:
You could mention the similarities between Flexbox `justify-center` and `align-items`.

Reply:
Actually, I have highlighted on those differences in my previous Flexbox article and given a link to it. I thought it would be off-topic to talk in detail about Flexbox in a Grid article.